### ISSUE TITLE
[bug] -rpath flag separator depends on build OS instead of host OS

### ISSUE THREAD

**Issue by Erlkoenig90** [conan-io/conan#7715]
### Environment Details (include every applicable attribute)
  * Operating System+version: Linux Mint 20 Ulyana 
  * Compiler+version: clang version 10.0.0, Target: x86_64-apple-darwin19
  * Conan version: 1.29.0 with PR #7695 applied
  * Python version: 3.8.2

### Steps to reproduce (Include if Applicable)
Install [osxcross](https://github.com/tpoechtrager/osxcross) on Linux, create a profile with `os=Macos` and attempt to build `glib` with [this PR](https://github.com/conan-io/conan-center-index/pull/2925) applied, like this:

### Logs (Executed commands with output) (Include/Attach if Applicable)
```sh
conan create . -pr:h x86_64-apple-darwin19-macosx-clang -pr:b default -o glib:with_elf=False
```
results in
```
[111/398] Linking target glib/libglib-2.0.0.dylib
FAILED: glib/libglib-2.0.0.dylib 
x86_64-apple-darwin19-clang  -o glib/libglib-2.0.0.dylib ... lots of stuff ... -Wl,-rpath=/home/niklas.guertler/.conan/data/pcre/8.41/_/_/package/ddd42dd21f101343926e8ee9dff966bdba84b078/lib /home/niklas.guertler/.conan/data/bzip2/1.0.8/_/_/package/d81a2c40ef3ca9d7c9333cd75ac1bcdef9183a77/lib/libbz2.a -Wl,-rpath=/home/niklas.guertler/.conan/data/bzip2/1.0.8/_/_/package/d81a2c40ef3ca9d7c9333cd75ac1bcdef9183a77/lib /home/niklas.guertler/.conan/data/zlib/1.2.11/_/_/package/099680e166ebfb4e7a7b9e0126a86c3d8bf1e7a7/lib/libz.a -Wl,-rpath=/home/niklas.guertler/.conan/data/zlib/1.2.11/_/_/package/099680e166ebfb4e7a7b9e0126a86c3d8bf1e7a7/lib -lgnuintl -liconv -liconv
ld: unknown option: -rpath=/home/niklas.guertler/.conan/data/pcre/8.41/_/_/package/ddd42dd21f101343926e8ee9dff966bdba84b078/lib
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```
The `libpcre.pc` file created by the `pkg_config` generator looks like this:
```
prefix=/home/niklas.guertler/.conan/data/pcre/8.41/_/_/package/ddd42dd21f101343926e8ee9dff966bdba84b078
libdir=${prefix}/lib
includedir=${prefix}/include

Name: pcre-libpcre
Description: Conan package: pcre-libpcre
Version: 8.41
Libs: -L${libdir} -lpcre  -Wl,-rpath="${libdir}"
Cflags: -I${includedir} -DPCRE_STATIC=1
Requires: bzip2 zlib
```
The `ld64` linker used by Clang to (cross-)compile for Mac OS doesn't understand rpath flags with a `=` separator, which is why the [`rpath_flags` function](https://github.com/conan-io/conan/blob/c829789c2cd96a888494e6b994de37b52d26e707/conans/client/build/compiler_flags.py#L31) uses `,` instead of `=` when compiling _on_ Mac OS. However, this does not work when cross-compiling on Linux _for_ Mac OS, resulting in the malformed `.pc` files.

I would suggest changing the `rpath_flags` function to take the host (instead of build) OS as parameter and use that to determine the `rpath_separator`. I will submit a PR that does that.

**Comment by jgsogo** [comment]
Hi, I'm afraid this change will work for this scenario because _"The `ld64` linker used by Clang to (cross-)compile for Mac OS doesn't understand rpath flags with a `=` separator"_, but it might be broken for other scenarios where the linker used by other compilers to cross-compile expect those files in the format natural for the machine where the compiler is running (the _build_ machine).

I'm not an expert on this topic, maybe @SSE4 has more sound opinion.

**Comment by Erlkoenig90** [comment]
Right, well it works for Cross-Compiling from Linux to Mac OS via osxcross and to Windows via MinGW32. I just noticed it's silly to pass `os_host` to `rpath_flags` because it's already in `settings`, so I reverted that. I added some logic to _only_ set the -rpath to `,` when compiling for Mac OS from Linux or Mac OS via a `GCC_LIKE`-compiler (but left the other solution as a comment, so we can decide). That should fix this issue without influencing other use cases.
